cds_ils/migrator/items/api.py

```diff
@@ -48,67 +48,43 @@ def set_document_pid(record):
 
 def import_items_from_json(dump_file, rectype="item"):
     """Load items from json file."""
-    model, provider = model_provider_by_rectype(rectype)
     with click.progressbar(json.load(dump_file)) as bar:
-        error_logger.error("ITEMS: PROCESSING {0}".format(dump_file))
-        migrated_logger.warning("ITEMS: PROCESSING {0}".format(dump_file))
+
         for record in bar:
+            log_extras = dict(
+                document_legacy_recid=record["id_bibrec"],
+                barcode=record["barcode"],
+            )
+
             click.echo(
                 'Importing item "{0}({1})"...'.format(
                     record["barcode"], rectype
                 )
             )
-
-            set_internal_location_pid(record)
-
             try:
+                set_internal_location_pid(record)
                 set_document_pid(record)
-            except DocumentMigrationError:
-                # there are items on the list which are not to be migrated
-                # if no document found
-                continue
-
-            # clean the item JSON
-            try:
+                # clean the item JSON
                 clean_item_record(record)
-            except ItemMigrationError as e:
-                click.secho(str(e), fg="red")
-                error_logger.error(
-                    "ITEM: {0} ERROR: {1}".format(record["barcode"], str(e))
+                new_item = import_record(
+                    record,
+                    rectype=rectype,
+                    legacy_id_key="barcode",
+                    log=log_extras,
                 )
-                continue
-
-            # check if the item already there
-            item = get_item_by_barcode(
-                record["barcode"], raise_exception=False
-            )
-            allow_updates = current_app.config.get(
-                "CDS_ILS_MIGRATION_ALLOW_UPDATES"
-            )
-            if item and not allow_updates:
-                click.secho(
-                    "Item {0}) already exists with pid: {1}".format(
-                        record["barcode"], item.pid
+            except Exception as exc:
+                handler = json_records_exception_handlers.get(exc.__class__)
+                if handler:
+                    handler(
+                        exc,
+                        document_legacy_recid=record.get("id_bibrec")
+                        or record.get("document_pid"),
```

Comment:
```suggestion
                        or record["document_pid"],
```

Reply:
in this case, for the loggers I prefer to have "None" than to fail, because that was the idea for logging